src/components/ImageView/index.native.js

```diff
@@ -41,6 +45,34 @@ class ImageView extends PureComponent {
         });
     }
 
+    componentDidMount() {
+        // Wait till animations are over to prevent stutter in navigation animation
+        this.state.interactionPromise = InteractionManager.runAfterInteractions(() => this.calculateImageSize());
```

Comment:
@aswin-s Do you recall the production steps for the animation bug here? We are looking to remove `InteractionManager.runAfterInteractions`.

Reply:
Wow this was over 2 years ago. If I remember correctly trying to calculate image size while animating the modal was causing frozen frames on android device. But this component has gone through multiple refactors after this change and even react-native-fast-image was forked to improve load performance. So this optimisation might not be required anymore.

Reply:
Thank you!